[Lulu Y] iP

luluyousef opened the pull request:
### **DukePro**

> “Your mind is for having ideas, not holding them.” – David Allen (source)


DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w4.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)
____
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

luluyousef committed:
Greet, Echo, and Exit

luluyousef committed:
Add and List

luluyousef committed:
Added ability to mark tasks as done.

luluyousef committed:
Added support for tracking todo, deadlines, and events

luluyousef committed:
Automated Text UI Testing

luluyousef committed:
Handled user input errors

luluyousef committed:
Added deleting task from the list.

luluyousef committed:
Complete level 7.

luluyousef committed:
Level 8

Joel-Sung commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+                f.createNewFile();
+            }
+            Scanner sc = new Scanner(f);
+            ArrayList<Task> t = new ArrayList<>();
```
I am not sure I like the name of the ArrayList. I think giving it a clearer name would improve the readability of your code. For example, 'taskList'.

Joel-Sung commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+            int ctr = 0;
+            Scanner scanner = new Scanner(System.in);
+            System.out.println("_______________________________________________");
+            System.out.println("Hello! I'm Duke\n" + "What can I do for you?");
```
I think placing a line break before the '+' would improve improve the readability of your code. This is because you add a next line character before the '+', so presenting your code in the same way the output will look would be nice.

Joel-Sung commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+                    System.out.println(d.getMessage());
+                }
+            }
+        }catch(IOException e){
```
I think you may have missed this. Add a space between the } and catch.

Joel-Sung commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+                    text = text.concat("0|");
+                }
+                text = text.concat(t.description + "|" + ((Event) t).by + "\n");
+            }
```
I think it is better to add a space around every 'if', 'else' and 'for' as this helps the readability of your code.

Joel-Sung commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+public class DukeException extends Exception{
```
I think you should add a space after Exception. I noticed that this is a recurring practice for your classes. Adding a space between the class declaration and the { will help to add readability to your code. I suggest you read the part on "White space within a statement" in the Java coding standard 😄 .

Joel-Sung commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task{
+
+    protected String by;
```
I think it would be better to keep your class variables as private. Do you have a reason to declare them as protected?

Joel-Sung reviewed (commented):
Overall, your code quality is pretty good in my opinion. Just some things to take note, I think you should start adding spaces within your statements as it is part of the Java coding standard. Maybe also consider declaring class variables as private, unless you have a reason to do otherwise.

tchiong commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task{
```
Could perhaps add header docs to the class!

tchiong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+            }
+            int ctr = 0;
+            Scanner scanner = new Scanner(System.in);
+            System.out.println("_______________________________________________");
```
Can perhaps use a function to print the dividers to make the code look more readable!

tchiong reviewed (commented):
Other than the comments from the other reviewers and myself, LGTM!

luluyousef committed:
Use More OOP

luluyousef committed:
Organize into package duke

luluyousef committed:
Add JUnit tests

luluyousef committed:
Package app as a JAR file

luluyousef committed:
Add JavaDoc comments

luluyousef committed:
Tweak Code to comply with coding standard

luluyousef committed:
Implement Level 9

luluyousef committed:
Task file

luluyousef committed:
tasks

luluyousef committed:
Merge branch 'branch-A-JavaDoc'

luluyousef committed:
Merge remote-tracking branch 'origin/add-gradle-support'

luluyousef committed:
Automate project builds using Gradle

luluyousef committed:
Use Checkstyle

luluyousef committed:
Add a GUI to Duke.

luluyousef committed:
The code does not have any assertions.

Certain Assumptions in the code needs to be verified.

Let's utilize Java Assertions and use the keyword assert to verify assumptions in the code.

luluyousef committed:
Merge pull request #2 from luluyousef/branch-A-Assertion

Branch A assertion

luluyousef committed:
Merge branch 'master' of https://github.com/luluyousef/ip

luluyousef committed:
Some methods in the code are too long in the Parser and Storage class.

Long methods are hard to read and affect code quality.

Let's follow the coding standards and break long methods by refractoring.

luluyousef committed:
Merge pull request #3 from luluyousef/branch-A-CodeQuality

Some methods in the code are too long in the Parser and Storage class.

luluyousef committed:
Merge branch 'master' of https://github.com/luluyousef/ip

luluyousef committed:
Improve the GUI

luluyousef committed:
Add C-Sort extension and fix loading issue in the Storage class

luluyousef committed:
Add a User Guide to the Project

luluyousef committed:
Add Ui Image

luluyousef committed:
Fix User Guide

luluyousef committed:
no message

luluyousef committed:
Fix DeleteCommand and Finalize Project
